test/e2e/page-objects/pages/dialog/create-contract.ts

```diff
@@ -0,0 +1,39 @@
+import { Driver } from '../../../webdriver/driver';
+
+class CreateContractModal {
+  protected driver: Driver;
+
+  private readonly confirmButtton = { text: 'Confirm', tag: 'button' };
+
+  private readonly cancelButton = { text: 'Cancel', tag: 'button' };
+
+  constructor(driver: Driver) {
+    this.driver = driver;
+  }
+
+  async check_pageIsLoaded(): Promise<void> {
+    try {
+      await this.driver.waitForMultipleSelectors([
+        this.confirmButtton,
+        this.cancelButton,
+      ]);
+    } catch (e) {
+      console.log(
+        'Timeout while waiting for create contract dialog to be loaded',
+        e,
+      );
+      throw e;
+    }
+    console.log('Create contract dialog was loaded');
+  }
+
+  async clickConfirm() {
+    await this.driver.clickElement(this.confirmButtton);
```

Comment:
same as above, to avoid a race condition in subsequent steps, we could use this method so we wait until the dialog is closed `clickElementAndWaitForWindowToClose`